src/librustc_mir/borrow_check/diagnostics/region_errors.rs

```diff
@@ -153,8 +153,6 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
         // Iterate through all the errors, producing a diagnostic for each one. The diagnostics are
         // buffered in the `MirBorrowckCtxt`.
 
-        // TODO(mark-i-m): Would be great to get rid of the naming context.
-        let mut region_naming = RegionErrorNamingCtx::new();
         let mut outlives_suggestion = OutlivesSuggestionBuilder::default();
```

Comment:
Can you reset `next_region_name` to 1 here?

Reply:
I thought about this, as it is the cause of the one test out change... but I'm not really sure we want to. It seems more correct to use the previous counter.

Reply:
I think I would prefer to have errors that we report be independent of each other.
The errors can potentially be reordered, and we will have to reset the counter and map between different item bodies anyway.

Reply:
Actually, looking at this again now. Won't `do_mirborrowck` only ever be called once on a MIR def id? In that case, the counter will reset for the next def we check in the constructor of `MirBorrowckCtxt`, right?

Reply:
@matthewjasper I just tried resetting the counter forcibly, and none of the test output changed. So I think my second comment is correct. The counter will be recreated starting at 1 each time we borrowck a new MIR def, and we do not need to reset manually.

Reply:
I guess I wasn't being clear with "we will have to reset the counter and map between different item bodies anyway." I really meant that we can't store the counter somewhere global and share it between borrow checks, so we also shouldn't persist it between errors.

I guess if this matters then someone will raise an issue.

Reply:
> I really meant that we can't store the counter somewhere global and share it between borrow checks, so we also shouldn't persist it between errors.

Ah I see what you mean now. I argue that we don't want to. Having consistent numbering between error messages on the same MIR body makes sense IMHO. Additionally, outlives suggestions will sometimes output a combined suggestion across errors, in which case we do want consistent naming across errors.